tests/Http/Event/CopyEventRequestHandlerTest.php

```diff
@@ -0,0 +1,889 @@
+use CultuurNet\UDB3\Model\ValueObject\Calendar\BookingAvailabilityType;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\DateRange;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\MultipleSubEventsCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Day;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Days;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Hour;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Minute;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\OpeningHour;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\OpeningHours;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\OpeningHours\Time;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\PeriodicCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\PermanentCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SingleSubEventCalendar;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\Status;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\StatusReason;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\StatusType;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SubEvent;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SubEvents;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\TranslatedStatusReason;
+use CultuurNet\UDB3\Model\ValueObject\Translation\Language;
+use DateTimeImmutable;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidFactoryInterface;
+
+class CopyEventRequestHandlerTest extends TestCase
+{
+    use AssertApiProblemTrait;
+
+    private MockObject $uuidFactory;
+    private CallableIriGenerator $iriGenerator;
+    private TraceableCommandBus $commandBus;
+    private CopyEventRequestHandler $copyEventRequestHandler;
+
+    private const ORIGINAL_EVENT_ID = '609a8214-51c9-48c0-903f-840a4f38852f';
+    private const NEW_EVENT_ID = '581ef86b-78bf-4083-ac88-a42d0f4dde87';
+
+    protected function setUp(): void
+    {
+        $this->commandBus = new TraceableCommandBus();
+        $this->commandBus->record();
+
+        $this->uuidFactory = $this->createMock(UuidFactoryInterface::class);
+        $this->uuidFactory->expects($this->any())
+            ->method('uuid4')
+            ->willReturn(Uuid::fromString(self::NEW_EVENT_ID));
+
+        $this->iriGenerator = new CallableIriGenerator(
+            fn (string $id) => 'https://mock.io/events/' . $id
+        );
+
+        $this->copyEventRequestHandler = new CopyEventRequestHandler(
+            $this->commandBus,
+            $this->uuidFactory,
+            $this->iriGenerator
+        );
+    }
+
+    /**
+     * @test
+     * @dataProvider validEventDataProvider
+     */
+    public function it_does_not_throw_when_given_valid_event_data($data, CopyEvent $expectedCommand): void
+    {
+        $response = $this->copyEventRequestHandler->handle(
+            (new Psr7RequestBuilder())
+                ->withBodyFromString(json_encode($data))
```

Comment:
Maybe use the "safer" method `Json::encode`?

Reply:
Yeah good point, I just copied this from `UpdateCalendarRequestHandlerTest`. I'll change it there as well.

Reply:
I updated all usages of `withBodyFromString()` where we used `json_encode()` inside it, to use the existing `withBodyFromArray()` instead. (Was probably added a bit later when we already had these usages of `withBodyFromString()` + `json_encode()`)

Reply:
This is now making a lot of tests fail because they use an `object` instead of `array` for the JSON data 🙃 I'll fix it.